workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Worker;
+using Improbable.Worker.Core;
+using Unity.Entities;
+using UnityEngine;
+using Entity = Unity.Entities.Entity;
+
+namespace Improbable.Gdk.Core
+{
+    public class Worker : IDisposable
+    {
+        public const long WorkerEntityId = -1337;
+
+        internal readonly Dictionary<EntityId, Entity> EntityMapping = new Dictionary<EntityId, Entity>();
+
+        public readonly Vector3 Origin;
+        public readonly string WorkerType;
+        public readonly string WorkerId;
+        public readonly World World;
+        public readonly Connection Connection;
+        public readonly ILogDispatcher LogDispatcher;
+
+        public delegate void WorkerEventHandler(Worker worker);
+        public static event WorkerEventHandler OnConnect;
+        public static event WorkerEventHandler OnDisconnect;
+
+        private Worker(ConnectionConfig config, ILogDispatcher logDispatcher, Vector3 origin)
+        {
+            Origin = origin;
+            WorkerId = config.WorkerId;
+            if (string.IsNullOrEmpty(WorkerId))
+            {
+                WorkerId = $"{config.WorkerType}-{Guid.NewGuid()}";
+            }
+            
+            Connection = ConnectionUtility.Connect(config, WorkerId);
+            LogDispatcher = logDispatcher;
+            WorkerType = config.WorkerType;
+            if (LogDispatcher is ForwardingDispatcher dispatcher)
+            {
+                dispatcher.SetConnection(Connection);
+            }
```

Comment:
Off topic for this PR - it might be better to add a virtual `OnConnected(Connection c)` method to `ILogDispatcher` with an empty impl such that we can call that indiscriminately on the dispatcher.

It also allows users to hook into this logic without messing with the Core

Reply:
Oh wait its an interface 😢 

Point still stands - just change to abstract class 😉

Reply:
yeah, I agree. would like to rework that a bit in a separate pr. There is still quite a bit of clean up left (e.g. the whole bootstrap)

Reply:
👍 Let's make tickets for these